Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/task/edit/presenter/EditTaskPresenter.java

```diff
@@ -0,0 +1,57 @@
+package com.novoda.todoapp.task.edit.presenter;
+
+import com.novoda.data.SyncedData;
+import com.novoda.event.DataObserver;
+import com.novoda.todoapp.navigation.Navigator;
+import com.novoda.todoapp.task.TaskActionListener;
+import com.novoda.todoapp.task.data.model.Task;
+import com.novoda.todoapp.task.edit.displayer.EditTaskDisplayer;
+import com.novoda.todoapp.task.presenter.IdProducer;
+import com.novoda.todoapp.tasks.service.TasksService;
+
+import rx.subscriptions.CompositeSubscription;
+
+public class EditTaskPresenter {
+
+    private final TasksService tasksService;
+    private final EditTaskDisplayer taskDisplayer;
+    private final IdProducer idProducer;
+
+    final TaskActionListener taskActionListener;
```

Comment:
looks like it's accessed directly from the tests. Make sense to include it in a package-protected ctor and instantiate the TaskActionListener in a static factory method that only exposes the other collaborators?

```java
public static EditTaskPresenter create(
    TasksService tasksService,
    EditTaskDisplayer taskDisplayer,
    Navigator navigator,
    IdProducer idProducer
) {
    TaskActionListener taskActionListener = new TaskActionListener(tasksService, taskDisplayer, navigator, idProducer);
    return new EditTaskPresenter(tasksService, taskDisplayer, idProducer, taskActionListener);
}

EditTaskPresenter(
    TasksService tasksService,
    EditTaskDisplayer taskDisplayer,
    IdProducer idProducer,
    TaskActionListener taskActionListener
) {
    this.tasksService = tasksService;
    this.taskDisplayer = taskDisplayer;
    this.idProducer = idProducer;
    this.taskActionListener = taskActionListener;
}

```